GoogleForCreators/web-stories-wp
# Animations: honor `prefers-reduced-motion` setting

## Context



Animated elements in a story are not visible at all when "Reduce Motion" setting is enabled in the OS settings.

## Summary



Changes an animated element's initial style (specifically, its `transform`) so that it's wrapped in a `@media (prefers-reduced-motion: no-preference)` media query, to prevent the animation from being placed outside the canvas and never getting moved (because animations are turned off).

## Relevant Technical Choices



* Uses `@media (prefers-reduced-motion: no-preference)` instead of `@media not (prefers-reduced-motion)` due to the JS AMP validator complaining
* Uses CSS custom properties for defining the initial transform in an inline style, and applying it in a media query in `<style amp-custom>`.

## To-do



N/A

## User-facing changes



Without Reduce Motion (unchanged behavior):

https://user-images.githubusercontent.com/841956/118848165-cacae380-b8ce-11eb-9e10-89bb00078380.mov

With Reduce Motion (no animation, but element visible):

https://user-images.githubusercontent.com/841956/118848154-c7375c80-b8ce-11eb-8614-6ffb81e43450.mov

## Testing Instructions



### QA






This PR can be tested by following these steps:

1. Add a few pages and elements to a story
2. Add animations to all of them
3. Preview story on frontend
4. Verify that animations play as expected, and that animated elements on next page aren't visible before animation starts
5. Turn on Reduce Motion in OS settings (see #7567 for steps)
6. Preview story again
7. Verify that elements are no longer animated, but visible at the some position they would be at after animations finish

### UAT







This PR can be tested by following these steps:

1.

## Reviews

### Does this PR have a security-related impact?



### Does this PR change what data or activity we track or use?



### Does this PR have a legal-related impact?



## Checklist




---



Fixes #7567
